### PR TITLE
3.x into Master

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 1]
```diff
@@ -260,17 +260,6 @@ file that was distributed with this source code.
...
-                        autocompleteInput.attr('data', data).append(data).trigger('change');
-
-                        // manually trigger the `select2:select` event
-                        autocompleteInput.attr('data', data).trigger({
-                            type: 'select2:select',
-                            params: {
-                                data: data
-                            }
-                        });
-
```
Comment (VincentLanglet) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:273]: There was a comment to remove this code, but I'm not sure about this.

@cecicifu Can you tell me what to do with this code on master ?

**Comment by VincentLanglet** [comment]
The BC layer won't be needed with https://github.com/symfony/symfony/pull/41493

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 2]
```diff
@@ -260,17 +260,6 @@ file that was distributed with this source code.
                           $('#{{ id }}_hidden_inputs_wrap').html('<input type="hidden" name="{{ full_name }}" value="'+xhr.responseJSON.objectId+'" />');
                         {% endif %}
 
-                        // append to Select2
-                        autocompleteInput.attr('data', data).append(data).trigger('change');
```
Comment (cecicifu) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:264]: Just remove attr function
`autocompleteInput.append(data).trigger('change');`

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 3]
```diff
@@ -260,17 +260,6 @@ file that was distributed with this source code.
...
-
-                        // manually trigger the `select2:select` event
-                        autocompleteInput.attr('data', data).trigger({
```
Comment (cecicifu) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:267]: Same here
`autocompleteInput.trigger({`

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig (hunk 1 above)
Comment (cecicifu) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:273]: Just remove attr function from lines 264 and 267.
264: `autocompleteInput.append(data).trigger('change');`
267: `autocompleteInput.trigger({`

**Comment by wouterj** [comment]
> The BC layer won't be needed with symfony/symfony#41493

I would keep it, as `getUsername()` will still be deprecated, so SonataAdminBundle will still receive deprecation notices.

**Comment by VincentLanglet** [comment]
I'm currently getting 
```
LogicException: You can not use the addFlash method if sessions are disabled. Enable them in "config/packages/framework.yaml".
```
errors.

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig (hunk 1 above)
Comment (VincentLanglet) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig]: Should be ok

**Review by cecicifu — APPROVED** [review]

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig (hunk 1 above)
Comment (cecicifu) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig]: I approved these changes, but i cannot review the others

**Comment by jordisala1991** [comment]
@VincentLanglet With Sf 5.3, session is no longer picked from container, it is picked from request->getSession().

You need to change tests that rely on doing a container->get('session') (to be compat with sf4.4 you should leave that line), but add a `request->setSession` line.

Similar to this: https://github.com/Runroom/runroom-packages/blob/master/packages/sortable-behavior-bundle/tests/Unit/SortableAdminControllerTest.php#L139-L166

**Comment by VincentLanglet** [comment]
Thanks @jordisala1991, it's fixed.

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**